Views/MainWindow.axaml.cs

```diff
@@ -21,251 +21,504 @@
-    /// <param name="e">The event arguments.</param>
-    private void OnActivated(object? sender, EventArgs e)
-    {
-        // Delegate focus and clipboard state updates to the MermaidEditorView
-        MermaidEditor.BringFocusToEditor();
-        MermaidEditor.UpdateClipboardStateOnActivation();
-    }
+        // Set up two-way synchronization between Editor and ViewModel
+        SetupEditorViewModelSync();
+
+        // Wire up clipboard and edit actions to ViewModel
+        WireUpClipboardActions();
 
-    #region Overrides
+        _logger.LogInformation("=== MainWindow Initialization Completed ===");
+    }
 
     /// <summary>
-    /// Handles additional cleanup and state management when the window is closed.
+    /// Sets the editor text, selection, and caret position while validating bounds and preventing circular updates.
     /// </summary>
-    /// <remarks>This override ensures that event handlers are unsubscribed and view model state is persisted
-    /// before the window is fully closed. It also resets internal state flags to prepare for future window operations.
-    /// Call the base method to ensure standard close behavior is preserved.</remarks>
-    /// <param name="e">An <see cref="EventArgs"/> instance containing the event data associated with the window close event.</param>
-    protected override void OnClosed(EventArgs e)
+    /// <param name="text">The text to set into the editor. Must not be <see langword="null"/>.</param>
+    /// <param name="selectionStart">Requested selection start index.</param>
+    /// <param name="selectionLength">Requested selection length.</param>
+    /// <param name="caretOffset">Requested caret offset.</param>
+    private void SetEditorStateWithValidation(string text, int selectionStart, int selectionLength, int caretOffset)
     {
+        _suppressEditorStateSync = true; // Prevent circular updates during initialization
         try
         {
-            UnsubscribeAllEventHandlers();
-            _vm.Persist();
+            Editor.Text = text;
+
+            // Ensure selection bounds are valid
+            int textLength = text.Length;
+            int validSelectionStart = Math.Max(0, Math.Min(selectionStart, textLength));
+            int validSelectionLength = Math.Max(0, Math.Min(selectionLength, textLength - validSelectionStart));
+            int validCaretOffset = Math.Max(0, Math.Min(caretOffset, textLength));
+            Editor.SelectionStart = validSelectionStart;
+            Editor.SelectionLength = validSelectionLength;
+            Editor.CaretOffset = validCaretOffset;
+
+            // Since this is yaml/diagram text, convert tabs to spaces for correct rendering
+            Editor.Options.ConvertTabsToSpaces = true;
+            Editor.Options.HighlightCurrentLine = true;
+            Editor.Options.IndentationSize = 2;
+            Editor.TextArea.IndentationStrategy = new MermaidIndentationStrategy(Editor.Options);
```

Comment:
I think your branch is stale and an old version. This `IndentationStrategy ` code has been in the code for a bit and shouldn't show as a new addition as part of this PR.